tests/test_macro.py

```diff
@@ -0,0 +1,186 @@
+import pytest
+import math
+import numpy as np
+
+from sampler import Sampler
+
+@pytest.fixture
+def sampler():
+    seed = '12345678901234567890abcdefghijklmnopqrstuvwxyz😊'
+
+    risk_limit = .25
+    contests = {
+        'Contest A': {
+            'winner': 60000,
+            'loser': 54000,
+            'ballots': 120000,
+            'numWinners': 1,
+        },
+        'Contest B': {
+            'winner': 30000,
+            'loser': 24000,
+            'ballots': 60000,
+            'numWinners': 1,
+        },
+        'Contest C': {
+            'winner': 18000,
+            'loser': 12600,
+            'ballots': 36000,
+            'numWinners': 1,
+        },
+    }
+
+    batches = {}
+    for i in range(200):
+        batches['Batch {}'.format(i)] = {
+            'Contest A': {
+                'winner': 200,
+                'loser': 180,
+                'ballots': 400,
+                'numWinners': 1
+            },
+        }
+
+        batches['Batch {} AV'.format(i)] = {
+            'Contest A': {
+                'winner': 100,
+                'loser': 90,
+                'ballots': 200,
+                'numWinners': 1
+            },
+        }
+
+    for i in range(100):
+        batches['Batch {}'.format(i)]['Contest B'] = {
+            'winner': 200,
+            'loser': 160,
+            'ballots': 400,
+            'numWinners': 1
+        }
+        batches['Batch {} AV'.format(i)]['Contest B'] =  {
+            'winner': 100,
+            'loser': 80,
+            'ballots': 200,
+            'numWinners': 1
+        }
+
+    for i in range(30):
+        batches['Batch {}'.format(i)]['Contest C'] = {
+            'winner': 200,
+            'loser': 140,
+            'ballots': 400,
+            'numWinners': 1
+        }
+        batches['Batch {} AV'.format(i)]['Contest C'] = {
+            'winner': 100,
+            'loser': 70,
+            'ballots': 200,
+            'numWinners': 1
+        }
+
+    for i in range(100, 130):
+        batches['Batch {}'.format(i)]['Contest C'] = {
+            'winner': 200,
+            'loser': 140,
+            'ballots': 400,
+            'numWinners': 1
+        }
+        batches['Batch {} AV'.format(i)]['Contest C'] = {
+            'winner': 100,
+            'loser': 70,
+            'ballots': 200,
+            'numWinners': 1
+        }
+
+
+    yield Sampler('MACRO', seed, risk_limit, contests, batches)
+
+
+
+def test_max_error(sampler):
+
+    # this is kind of a hacky way to do this but ¯\_(ツ)_/¯
```

Comment:
What's hacky with this? What would I do to make it not hacky?

Reply:
I think that comment was before I settled on a testing strategy. seems like there might have been a better way to set up the test data from Stark's paper, but this works. I can take the comment out if you think this is fine.